### PR TITLE
Patch lua CVE-2019-6706, CVE-2020-15888, nopatch CVE-2020-24342

### PR DESCRIPTION
**Pull request by dmcilvaney**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

Nopatch `lua` CVE-2020-24342. I was unable to reproduce both issues in 5.3.5, and the Ubuntu triage reports the same results [here](https://people.canonical.com/~ubuntu-security/cve/2020/CVE-2020-24342.html). This CVEs will need to be re-visited if we update to 5.4 since it is currently not addressed in any releases.
Applied a patch for `lua` CVE-2019-6706 from [here](http://lua.2524044.n2.nabble.com/CVE-2019-6706-use-after-free-in-lua-upvaluejoin-function-tt7685575.html)
Applied patches for CVE-2020-15888, CVE-20200-15945 from Open Embedded

###### Change Log  

- Nopatch `lua` CVE-2020-24342
- Patch `lua` CVE-2019-6706 
- Patch `lua` CVE-2020-15888
- Unify the patch comments on my previous Lua CVEs.

###### Does this affect the toolchain?  

**YES**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2020-15888
- https://nvd.nist.gov/vuln/detail/CVE-2020-24342
- https://nvd.nist.gov/vuln/detail/CVE-2019-6706 

###### Test Methodology

- `make input-srpms`
- validated the repo exploits for all CVEs (CVE-2020-15888 and CVE-2020-24342 was no-repro, while CVE-2019-6706 DID repro)
- local build of `lua`
- test of local build in progress
- test of new Lua package in VM **failed**. CVE-2020-15945 exploit still causes segfault. Dropping patch for CVE-2020-15945

### CODE REVIEW

**Comment by dmcilvaney** [comment]
Local toolchain & package build is in progress, I'll then re-validate the exploits against the new package RPM.

**Comment by dmcilvaney** [comment]
Looks like CVE-2020-15945 may be a nopatch as well, investigating and will add to this PR if so.

**Comment by dmcilvaney** [comment]
While researching CVE-2020-15945 I noticed that open embedded considers both CVE-2020-15888 and CVE-2020-15945 applicable to 5.3.5. I'm going to re-visit these.

File: SPECS/lua/CVE-2020-15888.nopatch [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# CVE-2020-15888 appears to not affect 5.3.5 (no repro of exploit)
+# NOTE: Patches needed if updating to 5.4:
```
Comment (dmcilvaney) [inline SPECS/lua/CVE-2020-15888.nopatch:2]: Revisit nopatch decision.

Comment (dmcilvaney) [reply SPECS/lua/CVE-2020-15888.nopatch]: Added patch for 15888 from Open Embedded.

**Review by PawelWMS — APPROVED** [review]

**Comment by dmcilvaney** [comment]
Exploit at http://lua-users.org/lists/lua-l/2020-07/msg00123.html still works unfortunately. Needs more investigation.

**Comment by dmcilvaney** [comment]
Rolling back fix for CVE-2020-15945, upstream patch has no affect on exploit. It is medium severity so can be re-visited later.

**Comment by dmcilvaney** [comment]
Rebased to latest 1.0-dev

**Review by schmittjoseph — APPROVED** [review]

**Pull request merged**